ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/result-filter/index.tsx

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import * as React from 'react'
+import { hot } from 'react-hot-loader'
+import styled from '../../styles/styled-components'
+
+import { Button, buttonTypeEnum } from '../button'
+
+import MarionetteRegionContainer from '../../container/marionette-region-container'
+import * as FilterBuilderView from '../../../component/filter-builder/filter-builder.view'
+import * as FilterBuilderModel from '../../../component/filter-builder/filter-builder'
+
+type Props = {
+  removeFilter: () => void
+  saveFilter: (transformToCql: () => any) => void
+  hasFilter: Boolean
+  resultFilter?: {}
+  isList?: Boolean
+}
+
+const Root = styled<Props, 'div'>('div')`
+  display: block;
+  padding: ${props => props.theme.minimumSpacing};
+`
+
+const EditorFooter = styled<Props, 'div'>('div')`
+  padding-top: 30px;
+  white-space: nowrap;
+
+  button {
+    display: inline-block;
+    width: ${props => (props.hasFilter && '49%') || '100%'};
+    ${props => props.hasFilter && 'margin-right: 1%;'};
+  }
+
+  .footer-remove {
+    display: ${props => (props.hasFilter && 'inline-block') || 'none'};
```

Comment:
We shouldn't be using the marionette pattern of showing or hiding things with css classes now that we are in react. I think you should change this:

```javascript
          <Button
            className="footer-remove"
            buttonType={buttonTypeEnum.negative}
            text="Remove Filter"
            onClick={this.props.removeFilter}
          />
          <Button
            className="footer-save"
            buttonType={buttonTypeEnum.positive}
            text="Save Filter"
            onClick={this.onSave.bind(this)}
          />
```

To a function called `getButtons()` that does conditional rendering.

Reply:
Targeting the styling with styled would be good here too instead of relying on classNames and button tags